src/lib.rs

```diff
@@ -228,7 +228,7 @@ pub(crate) fn read_file<P: AsRef<Path>>(path: P) -> ProcResult<String> {
 }
 
 pub(crate) fn write_file<P: AsRef<Path>, T: AsRef<[u8]>>(path: P, buf: T) -> ProcResult<()> {
-    let mut f = File::open(path)?;
+    let mut f = OpenOptions::new().write(true).open(path)?;
```

Comment:
Hah, that's an embarrassing bug.   I guess no one uses the write functionality.  I didn't personally add it, so I don't have a good sense of the use cases for this write functionality.  Also, clearly we have no automated tests for this (I"m going to open a new issue on that topic shortly)

Reply:
In general, I'd say that testing the write functionality is hard as writing to the writable files under /proc has system-wide effects. What I think could be done is running tests under a chroot with a fake /proc, could help make us more confident in the read tests too, but I'm not sure it'd be that easy to setup(not sure docker(or a similar container tool) can help here, unless it'd allow us to make fake files under /proc)

Reply:
Running tests under a chroot with a fake /proc is a clever idea, I like it!  Initially I had in mind a system where the tests are run under some qemu/kvm VM, which would let us write these values without worrying about impacting a real running system.  Running an actual VM is rather heavyweight, but it might be OK since I think the tests would be run perhaps once per release

Reply:
I think that running a VM could work as well, but I'm not sure how good the tools to instrument VMs for this kind of thing are, and I'm not sure everything would be easily inspectable that way.

I think having a deeper look into both approaches would be a good idea though.